csharp/src/Ice/LocalException.cs

```diff
@@ -308,7 +386,7 @@ public UnsupportedProtocolException(string reason, Protocol bad, Protocol suppor
             Supported = supported;
         }
 
-        public UnsupportedProtocolException(string reason, Protocol bad, Protocol supported, System.Exception ex) : base(reason, ex)
+        public UnsupportedProtocolException(string reason, Protocol bad, Protocol supported, Exception ex) : base(reason, ex)
```

Comment:
The only use of this exception is
```
throw new UnsupportedProtocolException("", protocol, Protocol.Ice1);
```

I think we should provide a constructor with out reason argument, and just have a custom message, we can also remove the other constructors that are never used.

Reply:
For a future PR.